File: CSharp/Shop/Ui/CLI/ConsoleUiTextCreator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Shop.Ui.CLI {
    using Items;
    using Customers;

    public class ConsoleUiTextCreator : UiTextCreator {
        public string ConstructMainMenu() {
            return "USER COMMANDS\n\nSHOP     - Shop Menu\nCUSTOMER - Customer Menu\nEXIT     - Exit Application\n\nINPUT: ";
        }
        public string ConstructShopMenu(List<ItemBatch> items) {
            string s = "SHOP ITEMS\n\n" + "Name                Price               Stock";

            if (items.Count > 0) {
                s += "\n";
            }

            for (int i = 0; i < items.Count; ++i) {
                s += items[i].Item.Name;
                int spaces = 20 - items[i].Item.Name.Length;
                for (int j = 0; j < spaces; ++j) {
                    s += " ";
                }
                s += items[i].Item.Value;
                spaces = 20 - items[i].Item.Value.ToString().Length;
                for (int j = 0; j < spaces; ++j) {
                    s += " ";
                }
                s += items[i].Stock;
            }

            s += "\n\nUSER COMMANDS\n\nBACK - Go back to the Main Menu\n\nINPUT: ";

            return s;
        }
        public string ConstructCustomerMenu(Customer customer) {
            string s = "YOUR ITEMS";
            s += "\n\n";
            for (int i = 0; i < customer.Items.Count; ++i) {
                s += customer.Items[i].Name;
            }
            s += "\n\n";
            s += "USER COMMANDS\n\nBUY    - Buy Item\nREFUND - Refund Item\nBACK   - Go back to the Main Menu\n\nINPUT: ";
            return s;
        }
        public string ConstructCustomerBuyMenu(List<ItemBatch> shopItems) {
            string s = "SHOP ITEMS\n\n";
            s += "Name                Price               Stock\n";
            for (int i = 0; i < shopItems.Count; ++i) {
                s += shopItems[i].Item.Name;
                int spaces = 20 - shopItems[i].Item.Name.Length;
                for (int j = 0; j < spaces; ++j) {
                    s += " ";
                }
                s += shopItems[i].Item.Value;
                spaces = 20 - shopItems[i].Item.Value.ToString().Length;
                for (int j = 0; j < spaces; ++j) {
                    s += " ";
                }
                s += shopItems[i].Stock;
                s += "\n";
            }
            s += "\nUSER COMMANDS\n\n";
            s+= "[ITEM]  - Buy item with the name [ITEM]\n";
            s += "BACK    - Go back to the Customer Menu\n\n";
            s += "INPUT: ";

            return s;
        }
        public string ConstructCustomerRefundMenu(List<Item> customerItems) {
            string s = "YOUR ITEMS\n\n";
            s += "Name                Value\n";
            for (int i = 0; i < customerItems.Count; ++i) {
                s += customerItems[i].Name;
                int spaces = 20 - customerItems[i].Name.Length;
                for (int j = 0; j < spaces; ++j) {
                    s += " ";
                }
                s += customerItems[i].Value;
                s += "\n";
            }
            s += "\nUSER COMMANDS\n\n[ITEM] - Refund item with [ITEM] name\nBACK   - Go back to the Customer Menu\n\nINPUT: ";
            return s;

        }
    }
}

File: Java/TddShop/src/Ui/ConsoleUiTextCreator.java
package Ui;
import Item.*;
import Customer.*;

import java.util.List;

public class ConsoleUiTextCreator {
    public String ConstructMainMenu(){
        String s ="USER COMMANDS\n\n";
        s += "SHOP     - Shop Menu\n";
        s += "CUSTOMER - Customer Menu\n";
        s += "EXIT     - Exit Application\n\n";
        s += "INPUT:";
        return s;
    }
    public String ConstructShopMenu(List<ItemBatch> items){
        String s ="SHOP ITEMS\n\n";
        s+="Name                Price               Stock\n";
        for(int i = 0; i < items.size(); ++i){
            int spacesAfterName = 20 - items.get(i).Item().Name().length();
            s += items.get(i).Item().Name();
            for(int j = 0; j < spacesAfterName; ++j){
                s +=" ";
            }
            String valueString = "" + items.get(i).Item().Value();
            int spacesAfterPrice = 20 - valueString.length();
            s += valueString;
            for(int j = 0; j < spacesAfterPrice; ++j){
                s += " ";
            }
            s += items.get(i).Stock;
            s += "\n";
        }
        s+="\nUSER COMMANDS\n\n";
        s+="BACK - Go back to the Main Menu\n\n";
        s+="INPUT:";
        return s;
    }
    public String ConstructCustomerMenu(Customer c){
        String s ="YOUR ITEMS\n\n";

        for(int i = 0; i < c.Items().size(); ++i){
            s += c.Items().get(i).Name();
            s+="\n";
        }
        s += "\n";
        s+="USER COMMANDS\n\n";
        s+="BUY    - Buy Item\n";
        s+="REFUND - Refund Item\n";
        s+="BACK   - Go back to the Main Menu\n\n";
        s+="INPUT:";
        return s;
    }
    public String ConstructCustomerBuyMenu(List<ItemBatch> shopItems){
        String s = "SHOP ITEMS\n\n";
        s += "Name                Price               Stock\n";

        for(int i = 0; i < shopItems.size(); ++i){
            int spaces = 20 - shopItems.get(i).Item().Name().length();
            s += shopItems.get(i).Item().Name();
            for(int j = 0; j < spaces; ++j){
                s += " ";
            }
            String itemValue = "" + shopItems.get(i).Item().Value();
            spaces = 20 - itemValue.length();
            s += itemValue;
            for(int j = 0; j < spaces; ++j){
                s += " ";
            }
            s += shopItems.get(i).Stock;
            s += "\n";
        }
        s += "\n";
        s += "USER COMMANDS\n\n";
        s+= "[ITEM]  - Buy item with the name [ITEM]\n";
        s+= "BACK    - Go back to the Customer Menu\n\n";
        s+="INPUT:";

        return s;
    }
    public String ConstructCustomerRefundMenu(List<Item> customerItems){
        String s = "YOUR ITEMS\n\n";
        s += "Name                Value\n";
        for(int i = 0; i < customerItems.size(); ++i){
            int spaces = 20 - customerItems.get(i).Name().length();
            s += customerItems.get(i).Name();
            for(int j = 0; j < spaces; ++j){
                s += " ";
            }
            s += customerItems.get(i).Value();
            s+="\n";
        }
        s+="\nUSER COMMANDS\n\n";
        s+="[ITEM] - Refund item with [ITEM] name\nBACK   - Go back to the Customer Menu\n\n";
        s+="INPUT:";
        return s;
    }
}
